lib/manageiq/util/file_splitter.rb

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env ruby
+#
+# Yes, you guessed it... this is basically the `split` command...
+#
+# The intent of this is to allow for greater flexibility and utility than is
+# provided by the BSD/GNU variants however, specifically allowing to pipe from
+# a pg_dump directly to a upload target (whether it be a mounted volume or
+# something like a FTP endpoint, though the former is basically already
+# supported with vanilla `split`).
+#
+# FTP, specifically, will be supported natively since ruby has Net::FTP support
+# built in (feature WIP).  This also allows for this to work correctly cross
+# platform, without having to be concerned with differences in `split`
+# functionality.
+
+require 'net/ftp'
+require 'optparse'
+
+class FileSplitter
+  attr_accessor :input_file, :byte_count, :ftp
+
+  def self.run(options = nil)
+    options ||= parse_argv
+    new(options).split
+  end
+
+  def self.parse_argv
+    options = {}
+    parser  = OptionParser.new do |opt|
+      opt.on("-b", "--byte-count=BYTES", "Number of bytes for each split") do |bytes|
+        options[:byte_count] = parse_byte_value(bytes)
+      end
+      opt.on("--ftp-host=HOST", "Host of the FTP server") do |host|
+        options[:ftp_host] = host
+      end
+      opt.on("--ftp-dir=DIR",   "Dir on the FTP server to save files") do |dir|
+        options[:ftp_dir] = dir
+      end
+    end.parse!
+
+    input_file, file_pattern = determine_input_file_and_file_pattern
+
+    options[:input_file]     = input_file
+    options[:input_filename] = file_pattern
+
+    options
+  end
+
+  def initialize(options = {})
+    @input_file     = options[:input_file] || ARGF
+    @input_filename = options[:input_filename]
+    @byte_count     = options[:byte_count] || 10_485_760
```

Comment:
This is basically defaulted to 10MB.  I am open to a better default (specifically for `pg_dump` outputs and backups), but I just went with something here that could be reasonable to deal with in the test suite, since I am testing this on the filesystem directly with 10MB temp files.